torch/_dynamo/symbolic_convert.py

```diff
@@ -341,6 +342,23 @@ def inner(self: "InstructionTranslatorBase", inst: Instruction):
                 self.jump(inst)
                 return
 
+            if isinstance(value, SymNodeVariable):
+                # if the assertion is normal shape expression.
+                # just install guard and bail out.
+                try:
+                    result = value.evaluate_expr()
```

Comment:
You don't want to do `evaluate_expr`. You want to get the underlying SymBool, and then `torch._check` on it. This will prevent unbacked SymInt from failing (and generate a runtime assert, as desired.)